frontends/web/src/routes/buy/moonpay.tsx

```diff
@@ -15,194 +15,179 @@
  * limitations under the License.
  */
 
-import { Component, createRef, ChangeEvent } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useState, useEffect } from 'react';
+import { useLoad } from '../../hooks/api';
+import { createRef, ChangeEvent } from 'react';
 import { IAccount } from '../../api/account';
+import { getConfig } from '../../api/backend';
+import { getMoonpayBuyInfo } from '../../api/exchanges';
 import Guide from './guide';
 import { Header } from '../../components/layout';
-import { load } from '../../decorators/load';
-import { translate, TranslateProps } from '../../decorators/translate';
 import { Spinner } from '../../components/spinner/Spinner';
 import { findAccount, getCryptoName } from '../account/utils';
 import { Button, Checkbox } from '../../components/forms';
 import { setConfig } from '../../utils/config';
 import A from '../../components/anchor/anchor';
 import style from './moonpay.module.css';
 
-type TBuyProps = {
+type TProps = {
     accounts: IAccount[];
     code: string;
 }
 
-type TLoadedBuyProps = {
-    moonpay: { url: string, address: string; };
-    config: any;
-}
-
-type State = {
-  agreedTerms: boolean;
-  height?: number;
-  iframeLoaded: boolean;
-}
+export const Moonpay = ({ accounts, code }: TProps) => {
+  const { t } = useTranslation();
+  const [agreedTerms, setAgreedTerms] = useState(false);
+  const [iframeLoaded, setIframeLoaded] = useState(false);
+  const [height, setHeight] = useState(0);
 
-type Props = TLoadedBuyProps & TBuyProps & TranslateProps;
+  const config = useLoad(getConfig);
+  const moonpay = useLoad(getMoonpayBuyInfo(code));
 
-class Moonpay extends Component<Props, State> {
-  public readonly state: State = {
-    agreedTerms: this.props.config.frontend.skipBuyDisclaimer,
-    iframeLoaded: false,
-  };
+  const account = findAccount(accounts, code);
+  const name = getCryptoName(t('buy.info.crypto'), account);
+  const ref = createRef<HTMLDivElement>();
+  let resizeTimerID: any;
 
-  private ref = createRef<HTMLDivElement>();
-  private resizeTimerID?: any;
+  useEffect(() => {
+    if (!!config) {
+      setAgreedTerms(config.frontend.skipBuyDisclaimer);
+    }
+  }, [config]);
 
-  public componentDidMount() {
-    window.addEventListener('resize', this.onResize);
-  }
+  useEffect(() => {
+    onResize();
+    window.addEventListener('resize', onResize);
 
-  public componentWillUnmount() {
-    window.removeEventListener('resize', this.onResize);
-  }
+    return () => window.removeEventListener('resize', onResize);
+  });
 
-  private handleSkipDisclaimer = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleSkipDisclaimer = (e: ChangeEvent<HTMLInputElement>) => {
     setConfig({ frontend: { skipBuyDisclaimer: e.target.checked } });
   };
```

Comment:
Keep in mind this should probably be specific to MoonPay terms, currently it looks like it would set config.frontend.skipBuyDisclaimer for all exchanges/brokers.

Reply:
Yes, I would change that inside Pocket Terms PR